src/Queue/TaskInterface.php

```diff
@@ -22,4 +24,12 @@ interface TaskInterface {
 	 */
 	public function run(array $data, int $jobId): void;
 
+	/**
+	 * Provides access to the DI container and its registered services
+	 *
+	 * @param \Cake\Core\ContainerInterface $container The DI container instance
+	 * @return void
+	 */
+	public function services(ContainerInterface $container): void;
```

Comment:
This sounds like we need to make it a major... 
I wonder if there is sth we could do before we major (for Cake5)

Reply:
Maybe we could just let users add a Trait (e.g. ServicesTrait) that could provide this as opt-in?